diff --git a/Makefile.tool.am b/Makefile.tool.am
index df95029..72fbada 100644
--- a/Makefile.tool.am
+++ b/Makefile.tool.am
@@ -16,7 +16,6 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_SEC_CAPS@ = \
 	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 endif
 
-TOOL_LDADD_COMMON = -lgcc
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
 	$(TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@) $(TOOL_LDADD_COMMON)
 if VGCONF_HAVE_PLATFORM_SEC
