diff --git a/libinotifytools/src/inotifytools.c b/libinotifytools/src/inotifytools.c
index 58089b1..56191fd 100644
--- a/libinotifytools/src/inotifytools.c
+++ b/libinotifytools/src/inotifytools.c
@@ -30,6 +30,7 @@
 #include <sys/ioctl.h>
 #include <sys/select.h>
 #include <sys/stat.h>
+#include <sys/statfs.h>
 #include <sys/types.h>
 #include <time.h>
 #include <unistd.h>
@@ -755,8 +756,8 @@ static const char* inotifytools_filename_from_fid(
 	int len = 0, name_len = 0;
 
 	// Match mount_fd from fid->fsid (and null fhandle)
-	fsid.info.fsid.val[0] = fid->info.fsid.val[0];
-	fsid.info.fsid.val[1] = fid->info.fsid.val[1];
+	fsid.info.fsid.__val[0] = fid->info.fsid.__val[0];
+	fsid.info.fsid.__val[1] = fid->info.fsid.__val[1];
 	fsid.info.hdr.info_type = FAN_EVENT_INFO_TYPE_FID;
 	fsid.info.hdr.len = sizeof(fsid);
 	watch* mnt = watch_from_fid(&fsid);
@@ -1267,7 +1268,7 @@ int inotifytools_watch_files(char const* filenames[], int events) {
 				return 0;
 			}
 			memcpy(&fid->info.fsid, &buf.f_fsid,
-			       sizeof(__kernel_fsid_t));
+			       sizeof(struct __fsid_t));
 
 			// Hash mount_fd with fid->fsid (and null fhandle)
 			int ret, mntid;
@@ -1283,8 +1284,8 @@ int inotifytools_watch_files(char const* filenames[], int events) {
 					free(dirname);
 					return 0;
 				}
-				fsid->info.fsid.val[0] = fid->info.fsid.val[0];
-				fsid->info.fsid.val[1] = fid->info.fsid.val[1];
+				fsid->info.fsid.__val[0] = fid->info.fsid.__val[0];
+				fsid->info.fsid.__val[1] = fid->info.fsid.__val[1];
 				fsid->info.hdr.info_type =
 				    FAN_EVENT_INFO_TYPE_FID;
 				fsid->info.hdr.len = sizeof(*fsid);
@@ -1606,8 +1607,8 @@ more_events:
 				memcpy((void*)&id, fid->handle.f_handle,
 				       sizeof(id));
 				printf("[fid=%x.%x.%lx;name='%s'] %s\n",
-				       fid->info.fsid.val[0],
-				       fid->info.fsid.val[1], id, name,
+				       fid->info.fsid.__val[0],
+				       fid->info.fsid.__val[1], id, name,
 				       filename ?: "");
 			}
 		}
diff --git a/libinotifytools/src/inotifytools/fanotify-dfid-name.h b/libinotifytools/src/inotifytools/fanotify-dfid-name.h
index ce00406..63de546 100644
--- a/libinotifytools/src/inotifytools/fanotify-dfid-name.h
+++ b/libinotifytools/src/inotifytools/fanotify-dfid-name.h
@@ -124,7 +124,7 @@ struct fanotify_event_info_header {
 /* Unique file identifier info record */
 struct fanotify_event_info_fid {
 	struct fanotify_event_info_header hdr;
-	__kernel_fsid_t fsid;
+	struct __fsid_t fsid;
 	/*
 	 * Following is an opaque struct file_handle that can be passed as
 	 * an argument to open_by_handle_at(2).
