--- accountsservice-22.07.5/meson.build.orig	2022-02-26 13:14:03.142152027 -0300
+++ accountsservice-22.07.5/meson.build	2022-02-26 13:15:09.964854058 -0300
@@ -101,8 +101,7 @@
 
   config_h.set('PATH_WTMP', '_PATH_WTMPX')
 else
-  path_wtmp = '/var/log/utx.log'
-  assert(run_command('test', '-e', path_wtmp).returncode() == 0, 'Do not know which filename to watch for wtmp changes')
+  path_wtmp = '/var/log/wtmp'
   config_h.set_quoted('PATH_WTMP', path_wtmp)
 endif
