--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -120,7 +120,10 @@
     if (${CMAKE_CXX_COMPILER} MATCHES  ".*mingw.*")
         target_link_libraries(graphite2 kernel32 msvcr90 mingw32 gcc user32)
     else (${CMAKE_CXX_COMPILER} MATCHES  ".*mingw.*")
-	target_link_libraries(graphite2 c gcc_s)
+	target_link_libraries(graphite2 c)
+	if (CMAKE_COMPILER_IS_GNUCXX)
+		target_link_libraries(gcc_s)
+	endif()
     endif()
     include(Graphite)
     if (BUILD_SHARED_LIBS)
