diff --git a/gcr/console-interaction.c b/gcr/console-interaction.c
index 47c2cb4..0a25c28 100644
--- a/gcr/console-interaction.c
+++ b/gcr/console-interaction.c
@@ -62,6 +62,10 @@ GType                  console_interaction_get_type    (void) G_GNUC_CONST;
 
 G_DEFINE_TYPE (ConsoleInteraction, console_interaction, G_TYPE_TLS_INTERACTION);
 
+#if !defined(_GNU_SOURCE) && !defined(_BSD_SOURCE)
+char *getpass(const char *);
+#endif
+
 #ifdef G_OS_WIN32
 /* win32 doesn't have getpass() */
 static gchar *
