diff --git a/gnu/obstack.c b/gnu/obstack.c
index 54b675d..223cba5 100644
--- a/gnu/obstack.c
+++ b/gnu/obstack.c
@@ -326,7 +326,7 @@ int obstack_exit_failure = EXIT_FAILURE;
 #   include <libio/iolibio.h>
 #  endif
 
-static _Noreturn void
+static __attribute_noreturn__ void
 print_and_abort (void)
 {
   /* Don't change any of these strings.  Yes, it would be possible to add
@@ -349,6 +349,6 @@ print_and_abort (void)
    variable by default points to the internal function
    'print_and_abort'.  */
 __attribute_noreturn__ void (*obstack_alloc_failed_handler) (void)
-  = print_and_abort;
+  = &print_and_abort;
 # endif /* !_OBSTACK_NO_ERROR_HANDLER */
 #endif /* !_OBSTACK_ELIDE_CODE */
